src/poetry/mixology/version_solver.py

```diff
@@ -42,16 +43,23 @@ class DependencyCache:
     def __init__(self, provider: Provider) -> None:
         self.provider = provider
         self.cache: dict[
-            tuple[str, str | None, str | None, str | None], list[DependencyPackage]
+            tuple[str, str | None, str | None, str | None, str | None],
+            list[DependencyPackage],
         ] = {}
-
-    @functools.lru_cache(maxsize=128)
-    def search_for(self, dependency: Dependency) -> list[DependencyPackage]:
+        self.search_for = functools.lru_cache(
+            maxsize=128
+            if metadata.version("poetry-core")  # type: ignore[no-untyped-call]
+            != "1.1.0a7"
```

Comment:
I don't like this but necessary evil since the hash for dependency is incorrect otherwise.